File: rpc/java/client/RcdClient.java
package rpc.java.client;

import com.codelab.rpc.rcd.Common;
import com.codelab.rpc.rcd.UserOuterClass;
import com.google.protobuf.util.JsonFormat;
import io.grpc.ManagedChannel;
import io.grpc.ManagedChannelBuilder;
import com.codelab.rpc.rcd.Service;
import com.codelab.rpc.rcd.RcdServiceGrpc;

/**
 * @author wangyanan
 */
public class RcdClient
{
    private final RcdServiceGrpc.RcdServiceBlockingStub stub;
    public RcdClient(String host, int port) {
        ManagedChannel channel = ManagedChannelBuilder.forAddress(host, port)
                .usePlaintext()
                .build();

        stub = RcdServiceGrpc.newBlockingStub(channel);
    }

    public String rec(int uid, int count) {
        Service.Request req = Service.Request.newBuilder()
                .setMethod(Common.Method.GET)
                .setUser(UserOuterClass.User.newBuilder().setId(uid))
                .setCount(count)
                .build();
        Service.Response rsp = stub.rec(req);

        String json = "";
        try {
            json = JsonFormat.printer().print(rsp);
            System.out.println("收到的数据: ");
            System.out.println(json);
        } catch (Exception e) {
            e.printStackTrace();
        }
        return json;
    }


    public static void main(String[] args) {
        System.out.println("job start...");
        RcdClient client = new RcdClient("localhost", 5100);
        String rsp = client.rec(1, 2);
        System.out.println("job done!!!");
    }
}

File: rpc/go/client/rcd_client.go
package main

import (
	"codelab.com/rpc/base"
	"codelab.com/rpc/rcd"
	"context"
	"time"

	"google.golang.org/grpc"
)

var log = base.Log
func main() {
	log.Infof("Connect the grpc Server in Go...")
	conn, err := grpc.Dial(":5300", grpc.WithInsecure())
	if err != nil {
		log.Fatalf("fail to dial: %v", err)
	}
	defer conn.Close()
	client := rcd.NewRcdServiceClient(conn)

	ctx, cancel := context.WithTimeout(context.Background(), 1*time.Second)
	defer cancel()
	req := &rcd.Request{
		Method: rcd.Method_GET,
		User:   &rcd.User{Id: 1},
		Count:  2,
	}
	str, _ := base.Proto2Json(req)
	log.Infof("request: %s", str)

	rsp, err := client.Rec(ctx, req)
	if err != nil {
		log.Infof("%s.Rec(_) = _, %s: ", client, err)
	}
	str, _ = base.Proto2Json(rsp)
	log.Infof("response: %s", str)
	log.Infof("client done....")
}
